github-actions/rebase-pr/entrypoint.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+set -e
+
+# Set email and name config
+git config --global user.email "devops+rebase@angular.io";
+git config --global user.name "Angular Rebase Bot";
+
+# Exit code for a successful run
+SUCCESS_EXIT_CODE=0;
+
+# The repository name and branch name for the PRs head and base
+PR_REPOSITORY=$(jq -r ".pull_request.head.repo.full_name" "$GITHUB_EVENT_PATH");
+PR_BRANCH=$(jq -r ".pull_request.head.ref" "$GITHUB_EVENT_PATH");
+TARGET_REPOSITORY=$(jq -r ".pull_request.base.repo.full_name" "$GITHUB_EVENT_PATH");
+TARGET_BRANCH=$(jq -r ".pull_request.base.ref" "$GITHUB_EVENT_PATH");
+
+# Change to the github workspace and initialize the repo if needed.
+cd "$GITHUB_WORKSPACE";
+git init;
+
+# Add pr and target repos as remotes, then fetch the remote branches
+git remote add pr https://x-access-token:$TOKEN@github.com/$PR_REPOSITORY;
+git remote add target https://x-access-token:$TOKEN@github.com/$TARGET_REPOSITORY;
+git fetch pr "$PR_BRANCH";
+git fetch target "$TARGET_BRANCH";
+
+# Checkout the PR change branch from its remote
+git checkout --track pr/"$PR_BRANCH"
```

Comment:
Optional nit: Ideally, there would be a comment explaining why we use `--track` here, and below why we use `--force-with-lease` (just for later reference), but that can be a follow-up :smile: